astropy/io/fits/hdu/image.py

```diff
@@ -1198,6 +1224,8 @@ def _verify(self, option='warn'):
 class _IndexInfo:
     def __init__(self, indx, naxis):
         if _is_int(indx):
+            if indx < 0:
+                indx = indx + naxis
```

Comment:
Can we add a comment about what's going on here?

I didn't dive in to figure out how `_IndexInfo` is even used -- assume this has to do with negative indexing and needing to wrap around, but it seems strange to me that this wasn't supported before